server/market/market.go

```diff
@@ -614,6 +688,10 @@ func (m *Market) SubmitOrderAsync(rec *orderRecord) <-chan error {
 	select {
 	case <-m.running:
 	default:
+		if rec.order.Type() == order.CancelOrderType {
+			err := m.processCancelOrderWhileSuspended(rec)
```

Comment:
I'm not sure if this is ever a problem, but the function this is in is `SubmitOrderAsync` and this has the potential to block for a bit with two db interactions.

Reply:
It's not a problem now since `SubmitOrderAsync` is only called from `SubmitOrder` which blocks, but it's definitely not the intention for this function to block, so I'll update it.